pandas/tests/indexing/test_coercion.py

```diff
@@ -466,8 +466,6 @@ def test_insert_index_datetimes(self, fill_val, exp_dtype):
         with pytest.raises(TypeError, match=msg):
             obj.insert(1, 1)
 
-        pytest.xfail("ToDo: must coerce to object")
```

Comment:
i think this indicates something that we need to fix.  maybe should write out the assertion so that the xfail is meaningful

Reply:
Thanks @jbrockmendel, I was planning on pinging you and @WillAyd after a bit more digging into this, but you beat me to it :). This was originally changed in #18721 from a comment to the xfail that exists now. The comments were originally added 
in https://github.com/pandas-dev/pandas/commit/7c0b74287408a2dcff4cbcafbbebbf9847a6905a. I'm not sure what should be asserted here; from the history it seems like it refers to the tests above.

Reply:
Also note those comments still exist in the test immediately below.

Reply:
if we want to coerce to object (and i think we do), then for starters the insert on L467 shouldn't raise but should instead be equivalent to `obj.astype(object).insert(1, 1)`.  so i think can just change that and then xfail it?

the harder question is what we do in the two cases above that with mismatched tz or mismatched tzawareness.  with mismatched tz could just cast (#37605), with mismatched tzawareness could also cast to object.

Reply:
if you can create an issue for this in followup (and PR if you can!)